### PR TITLE
Support creating a new maven module to existing multi-module project via 'New Module...' command

### LINKED ISSUE #849: add new project as module of existing project
We can provide an extra entry for this feature, "Add a module" from a project node.

E.g. Similar as mentioned in this doc: https://www.codejava.net/ides/intellij/create-multi-module-maven-project-intellij

### PR DESCRIPTION
**Pull request by testforstephen**
Closes #849 

What this PR does:
1) In File Explorer context menu, rename the old menu `"Create Maven Project"` to `"Maven / New Project..."`
2) Add a new context menu `"Maven / New Module..."` to the File Explorer view. This command will provide a simplified wizard to create a new module to the existing maven project.
image

3) Automatically import the newly created maven module as a Java project after the wizard "New Project/Module" is finished.

The difference between "New Project..." and "New Module...":
- "New Project..." command is a general command to use maven archetype to create a maven project in any location, either inside or outside your current workspace.
- The “New Module…” command is a convenient way to create a new module within your current workspace. It prompts you to select the parent pom and automatically adds the new module to it.

See the demo on "New Module..." command.

https://github.com/microsoft/vscode-maven/assets/14052197/4aac6a4d-4d9d-4a59-8096-d2e7231c936c

### CODE REVIEW

File: src/archetype/createProject/SelectParentPomStep.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        const disposables: Disposable[] = [];
+        try {
+            return await new Promise<StepResult>((resolve) => {
+                const pickBox: QuickPick<ParentPomPickItem> = window.createQuickPick<ParentPomPickItem>();
+                pickBox.title = metadata.title;
+                pickBox.placeholder = "Select the parent...";
+                pickBox.matchOnDescription = true;
+                pickBox.ignoreFocusOut = true;
+                pickBox.items = items;
```
Comment (jdneo) [inline src/archetype/createProject/SelectParentPomStep.ts:40]: Will it be better if we skip this step when there is only one available item?

File: src/archetype/ArchetypeModule.ts [hunk 2]
```diff
@@ -158,17 +202,32 @@ async function createBasicMavenProject(metadata: IProjectCreationMetadata): Prom
...
         const targetUri = vscode.Uri.joinPath(vscode.Uri.file(targetFolder), artifactId);
         await workspace.fs.copy(templateUri, targetUri, { overwrite: true });
 
         // update groupId/artifactId in pom.xml
-        p.report({ message: "Updating pom.xml file...", increment: 20 });
+        p.report({ message: "Updating pom.xml file...", increment: 10 });
         const pomUri = vscode.Uri.joinPath(targetUri, "pom.xml");
         let pomContent = (await workspace.fs.readFile(pomUri)).toString();
+        let parentPom = "";
+        let compilerSource = "17";
```
[the remark this answers is not in the document]
Comment (testforstephen) [reply src/archetype/ArchetypeModule.ts]: nice

File: src/archetype/createProject/SelectParentPomStep.ts (hunk 1 above)
Comment (testforstephen) [reply src/archetype/createProject/SelectParentPomStep.ts:40]: We always add a `<None>` in the pick items. If only `<None>` is left in the items, I prefer to keep it to make user know there is no parent pom in current workspace.

Comment (jdneo) [reply src/archetype/createProject/SelectParentPomStep.ts:40]: ok

**Comment by jdneo** [comment]
Would you like to register the new module command to the Java Projects view as well?

It's not necessary to do that in this PR. If you plan to do that, an issue can be filed first to track it.

**Comment by testforstephen** [comment]
> Would you like to register the new module command to the Java Projects view as well?
> 
> It's not necessary to do that in this PR. If you plan to do that, an issue can be filed first to track it.

Issue created https://github.com/microsoft/vscode-maven/issues/1019. Yes, I plan to add this feature to the Java Projects view as well. For now, I will release it in vscode-maven first to test its user adoption.

**Review by jdneo — APPROVED** [review]
LGTM

**Pull request merged**